src/emittermixin.js

```diff
@@ -225,6 +225,7 @@ const EmitterMixin = {
 	 * @param {String} event The name of the event.
 	 * @param {...*} [args] Additional arguments to be passed to the callbacks.
 	 * @method core.EmitterMixin#fire
+	 * @returns {*} Value returned by the last callback processed in the event loop.
```

Comment:
Let's not do this change in this PR. Please use the existing emitter API. You can implement the `checkEnabled` event in this way:

```js
const data = { isEnabled: true };
fire( 'checkEnabled', data );

if ( data.isEnabled ) ....
```